x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/TransportSamlInitiateSingleSignOnAction.java

```diff
@@ -140,4 +123,23 @@ private void buildUserFromAuthentication(SecondaryAuthentication secondaryAuthen
             }
         );
     }
+
+    private void possiblyReplyWithSamlResponse(SamlAuthenticationState authenticationState, String statusCode, Exception e,
```

Comment:
Since this only handles a failure response, I think it's worth naming it accordingly.

```suggestion
    private void possiblyReplyWithSamlFailure(SamlAuthenticationState authenticationState, String statusCode, Exception e,
```